### PR TITLE
Created new Spice: BaconIpsum

### PR DESCRIPTION
**Pull request by puskin**
**What does your Instant Answer do?**

Returns a variable number of BaconIpsum Paragraph

**What problem does your Instant Answer solve (Why is it better than organic links)?**

https://duck.co/ideas/idea/24/bacon-ipsum-powered-by-baconipsum-com

**What are some example queries that trigger this Instant Answer?**

baconipsum 3

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**

WebDesigners

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**

https://duck.co/ideas/idea/24/bacon-ipsum-powered-by-baconipsum-com

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

bacon
## Checklist

Please place an 'X' where appropriate.

```
[X] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
[X] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
[X] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
[] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
[X] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        [] Google Chrome
        [] Firefox
        [] Opera
        [] IE 10

    - Windows 7
        [] Google Chrome
        [] Firefox
        [] Opera
        [] IE 8
        [] IE 9
        [] IE 10

    - Windows XP
        [] IE 7
        [] IE 8

    - Mac OSX
        [X] Google Chrome
        [X] Firefox
        [] Opera
        [X] Safari

    - iOS (iPhone)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - iOS (iPad)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - Android
        [] Firefox
        [] Native Browser
        [] Google Chrome
        [] Opera
```

### CODE REVIEW

**Comment by jagtalon** [comment]
Hey, @puskin94!

- Would be great if we could see 4 paragraphs instead of just one big one:
screen shot 2015-01-05 at 4 33 05 pm

- It looks like it doesn't work with just "bacon ipsum"
screen shot 2015-01-05 at 4 34 21 pm

- I think it would be good if we use the 'info' template here so that we have a "show more" thing like in https://duckduckgo.com/?q=green+day+artist&ia=music

screen shot 2015-01-05 at 4 35 44 pm

Here's the code for that: https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/share/spice/lastfm/artist/lastfm_artist.js#L23

Tell me what you think!

**Comment by puskin94** [comment]
Hey @jagtalon ,
I updated a little bit the code, now it works without param ( if empty, it will be set to 1 ) and all the paragraphs are spaced. Like:
schermata 2015-01-06 alle 17 08 45
Actually, I prefer the text layout for this IA :D Let me know!

File: share/spice/bacon_ipsum/bacon_ipsum.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+(function (env) {
+    "use strict";
+    env.ddg_spice_bacon_ipsum = function(api_result){
+
+
+        if (api_result.error) {
```
Comment (jagtalon) [inline share/spice/bacon_ipsum/bacon_ipsum.js]: Would also check if `api_result` exists, i.e.,

```javascript
if (!api_result || api_result.error) {
   ...
}
```

**Comment by jagtalon** [comment]
@puskin94 I like it! I'm just not sure about the tab "Bacon Says ..."--it looks too informal (although you could argue that the IA itself is meant to be funny.) What do you think @mattr555 @MrChrisW @abeyang?

**Comment by puskin94** [comment]
@jagtalon Should I change it to "BaconIpsum" ??

**Comment by jagtalon** [comment]
"Bacon Ipsum" sounds good :)

**Comment by puskin94** [comment]
@jagtalon Done! Thanks for the feedback!

**Comment by jagtalon** [comment]
:+1:

**Comment by jagtalon** [comment]
@puskin94 it's live! https://duckduckgo.com/?q=bacon+ipsum&ia=baconipsum

**Pull request merged**